x-pack/legacy/plugins/ml/public/services/http_service.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// service for interacting with the server
+
+import chrome from 'ui/chrome';
+
+// @ts-ignore
+import { addSystemApiHeader } from 'ui/system_api';
+import { fromFetch } from 'rxjs/fetch';
+import { from, Observable } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+
+export interface HttpOptions {
+  url?: string;
+}
+
+function getResultHeaders(headers: HeadersInit): HeadersInit {
+  return addSystemApiHeader({
+    'Content-Type': 'application/json',
+    'kbn-version': chrome.getXsrfToken(),
+    ...headers,
+  });
+}
+
+export function http(options: any) {
+  return new Promise((resolve, reject) => {
+    if (options && options.url) {
+      let url = '';
+      url = url + (options.url || '');
+      const headers: Record<string, string> = addSystemApiHeader({
+        'Content-Type': 'application/json',
+        'kbn-version': chrome.getXsrfToken(),
+        ...options.headers,
+      });
+
+      const allHeaders =
+        options.headers === undefined ? headers : { ...options.headers, ...headers };
+      const body = options.data === undefined ? null : JSON.stringify(options.data);
+
+      const payload: RequestInit = {
+        method: options.method || 'GET',
+        headers: allHeaders,
+        credentials: 'same-origin',
+      };
+
+      if (body !== null) {
+        payload.body = body;
+      }
+
+      fetch(url, payload)
+        .then(resp => {
+          resp.json().then(resp.ok === true ? resolve : reject);
+        })
+        .catch(resp => {
+          reject(resp);
+        });
+    } else {
+      reject();
+    }
+  });
+}
+
+export function httpCall<T>(url: string, options: RequestInit): Observable<T> {
+  const requestInit: RequestInit = {
+    ...options,
+    credentials: 'same-origin',
+    method: options.method || 'GET',
+    ...(options.body ? { body: JSON.stringify(options.body) } : {}),
+    headers: getResultHeaders(options.headers ?? {}),
+  };
+
+  return fromFetch(url, requestInit).pipe(
```

Comment:
Would we want to add a `catchError` for stuff like network errors?

Reply:
We don't need to. When you subscribe to Observable you can also add a callback for `.catch` in the same manner as catching errors with Promises. 
`catchError` is used to handle an error and pass the new observable to the observable sequence, so in that case, the error will end up in subscribe (and we do not want it for all HTTP calls :) )